src/balancer/BalancerController.sol

```diff
@@ -235,7 +248,10 @@ contract BalancerController is IController {
     {
         uint steps = swaps.length;
         for (uint i; i < steps - 1; i++) {
-            if (swaps[i].assetOutIndex != swaps[i+1].assetInIndex)
+            if (
+                swaps[i].assetOutIndex != swaps[i+1].assetInIndex ||
+                swaps[i+1].amount > 0
+            )
```

Comment:
This seems like a solid solution vs the one I suggested (basically you are blocking all swaps with intermediate values rather than calculating those values). Seems simpler and fits the vast majority of use cases.